frb/igm.py

```diff
@@ -153,8 +153,70 @@ def z_from_DM(DM, cosmo=None, coord=None, corr_nuisance=True):
     # Return
     return z
 
+def f_diffuse(z, cosmo = Planck15, return_rho = False):
+  """
+  Calculate the cosmic fraction of baryons
+  in diffuse gas phase based on our empirical
+  knowledge of baryon distributions and their 
+  ionization state.
 
-def average_DM(z, cosmo=None, cumul=False, neval=10000, mu=1.3):
+  Args:
+    z: float or ndarray
+      Redshift
+    cosmo: astropy Cosmology, optional
+      Cosmology of the universe.
+    return_rho: bool, optional
+      If true, the diffuse gas density
+      is returned too.
+  Returns:
+    f_diffuse: float, ndarray
+      Diffuse gas baryon fraction.
+    rho_diffuse: astropy quantity
+      Physical diffuse gas density.
+      Returned if return_rho
+      is set to true.
+  """
+  # Get comoving baryon mass density
+  rho_b = cosmo.Ob0 * cosmo.critical_density0.to('Msun/Mpc**3')
+
+  # Dense components
+  rho_Mstar = avg_rhoMstar(z, remnants=True)
+  rho_ISM = avg_rhoISM(z)
+
+  # Diffuse gas fraction
+  f_diffuse = 1 - ((rho_Mstar+rho_ISM)/rho_b).value
+  if not return_rho:
+    return f_diffuse
+  else:
+    return f_diffuse, rho_b*f_diffuse*(1+z)**3
+
+def ne_cosmic(z, cosmo = Planck15, mu = 4/3):
+  """
+  Calculate the average cosmic electron
+  number density as a function of redshift.
+  Args:
+    z: float or ndarray
+      Redshift
+    cosmo: astropy Cosmology, optional
+      Cosmology in which the calculations
+      are to be performed.
+  Returns:
+    ne_cosmic: astropy Quantity
+      Average number density of
```

Comment:
comoving or physical?

Reply:
Physical. Just updated the docstring to reflect this,